extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -488,8 +539,14 @@ export class AmpA4A extends AMP.BaseElement {
                   if (isValid) {
                     return creative;
                   }
-                  return Promise.reject(
-                      'Key failed to validate creative\'s signature.');
+                  // Only report if signature is expected to match, given that
+                  // multiple key providers could have been specified.
+                  if (verifyHashVersion(signature, keyInfo)) {
```

Comment:
Added logging statement within promise chain when we had a signature but validation failed against all providers.  This will cause multiple error statements to potentially occur (should there be one key provider that matches the signature hash but failed validation) but cannot see easy way around it.